examples/ruby/ruby_client.rb

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env ruby
+# confirmed for ruby versions 2.6+
+
+# Load the gem
+require 'openapi_client'
+
+CERT_DIR = 'config/https'
+SSL_CERT_FILE = 'ca.crt'
+CONJUR_CERT_FILE = 'conjur.crt'
+CONJUR_KEY_FILE = 'conjur.key'
+
+ACCOUNT = "dev"
+LOGIN = "admin"
+ADMIN_API_KEY = ENV["CONJUR_ADMIN_API_KEY"]
+if ADMIN_API_KEY.empty?
+  STDERR.puts "Conjur Admin API key Environment Variable not set!"
+  STDERR.puts "Use the privided script, or be sure to first execute: "
+  STDERR.puts 'export CONJUR_ADMIN_API_KEY="$(docker-compose exec conjur conjurctl role retrieve-key dev:user:admin | tr -d \'\\r\')'
+  exit(1)
+end
+
+# Constants
+new_password = "N3w-Passw0rd!"
+secret = "supersecretstuff"
+secret_id = "sampleSecret"
+empty_policy = IO.read("examples/config/simple.yml")
+policy = IO.read("examples/config/policy.yml")
+
+# Setup client configuration
+config = OpenapiClient.configure
+# config.debugging = true
+config.scheme = "https"
+config.host = "localhost"
+
+config.username = LOGIN
+config.password = ADMIN_API_KEY
+
+config.verify_ssl = true
+config.ssl_ca_cert = File.join(CERT_DIR, SSL_CERT_FILE)
+config.cert_file = File.join(CERT_DIR, CONJUR_CERT_FILE)
+config.key_file = File.join(CERT_DIR, CONJUR_KEY_FILE)
+
+# Authenticating admin using basicAuth
+authn_instance = OpenapiClient::AuthnApi.new
+token = nil
+puts "Authenticating admin..."
+begin
+  token = authn_instance.authenticate(account=ACCOUNT, login=LOGIN, 
+                                      body=ADMIN_API_KEY, opts={accept_encoding: 'base64'})
+  puts "Base64 encoded token: #{token}"
+rescue OpenapiClient::ApiError => e
+  puts "Exception when calling AuthnApi->authenticate: #{e}"
+  exit(1)
+end
+
+# Change admin password using basicAuth
+puts
+puts "Changing admin password..."
+begin
+  authn_instance.set_password(account=ACCOUNT, body=new_password)
+  puts "Password change successful."
+rescue OpenapiClient::ApiError => e
+  puts "Exception when calling AuthnApi->set_password: #{e}"
+  exit(1)
+end
```

Comment:
Not a blocker but both python and ruby do exits on exceptions. I wonder if we should just avoid wrapping any of these in try/rescues and try/excepts